lib/impure/db_sqlite.nim

```diff
@@ -716,8 +717,63 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
 proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
   discard finalize(sqlPrepared.PStmt)
 
+template dbBindParamError*(paramIdx: int,val:untyped) =
+  ## Raises a `DbError` exception.
+  var e: ref DbError
+  new(e)
+  e.msg = "error binding param in position " & $paramIdx 
+  raise e
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int32) =
+  ## Binds a int32  to the specified paramIndex.
+  if SQLITE_OK != bind_int(ps.PStmt, paramIdx.int32, val):
+    dbBindParamError(paramIdx,val)
+
+when sizeof(int) == 4:
+  proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int) =
+    ## Binds a int32  to the specified paramIndex.
+    if SQLITE_OK != bind_int(ps.PStmt, paramIdx.int32, val.int32):
+      dbBindParamError(paramIdx,val)
```

Comment:
You could have just implement this so that it calls `bindParam(val.int32)` when `sizeof(int) == 4` and `bindParam(val.int64)` otherwise...

Reply:
ok, I checked ormin , it always using bind int64

Reply:
cstring cant store compressed data right? since there's many zero terminers inside the data.
so the binding api should have pass string version